shared/utils/conversion.ts

```diff
@@ -0,0 +1,15 @@
+export function convertScale(rating: number): number;
+export function convertScale(rating: number | undefined): number | undefined;
+export function convertScale(rating: number | undefined) {
+  if (rating === undefined) return undefined;
+  else if (rating < 0) return rating;
+  return rating ** 2;
+}
+
+export function revertScale(rating: number): number;
+export function revertScale(rating: number | undefined): number | undefined;
+export function revertScale(rating: number | undefined) {
+  if (rating === undefined) return undefined;
+  else if (rating < 0) return rating;
+  return Math.sqrt(rating);
+}
```

Comment:
Same goes for revertScale
```suggestion
export const revertScale = (rating: number) =>
  rating < 0 ? rating : Math.sqrt(rating);
```